WordPress/gutenberg
# `InputControl`: Fix acceptance of falsy values in controlled updates

## What?
Makes `InputControl` handling of updates from props distinct from any other action.

## Why?
To fix #42455. Presently the `RESET` action is used to handle updates from props and that favors a fallback value over a falsy one:
https://github.com/WordPress/gutenberg/blob/875a20481d2329adfb4eb87fe082dce24f2e0f8c/packages/components/src/input-control/reducer/reducer.ts#L98-L101
That means that that falsy values coming from props will not be used which is the cause of the issue.

## How?
- In order to have a test that will fail in case of another regression, an existing test is updated to use a falsy value
- The handling of updates from props made distinct to any other existing action types

## Testing Instructions
1. After fetching and switching to this branch, checkout the first commit
2. run `npm test-unit input-control` to verify the changed test fails
3. Switch back to the whole branch `git switch -`
4. repeat step 2 but note the test passes
5. Try reproducing the issue in #42455 as detailed in its reproduction instructions
